HenBot/GelbooruSourceService.cs

```diff
@@ -0,0 +1,24 @@
+﻿using RestSharp;
+
+namespace HenBot
+{
+    public class GelbooruSourceService
+    {
+        static readonly RestClient _client = RestHttpClient.GetRestClient();
+
+        public static async Task<List<Post>> GetPostsAsync(int limit, string tags, Ratings rating, int page)
+        {
+            var request = new RestRequest($"index.php?page=dapi&s=post&q=index&limit={limit}&tags={TagsQueryBuilder.BuildTagsQuery(rating, tags)}&pid={page}&json=1");
```

Comment:
It's a much better way to build your requests using RestSharp build-in .AddParameter method. This way you don't have (and you shouldn't) escape any query parameters special characters by your own.

If you apply this style of creating requests, you should rewrite TagsQueryBuilder.BuildTagsQuery and replace all escaped characters using [this manual](https://www.w3schools.com/tags/ref_urlencode.ASP)

```suggestion
            var request = new RestRequest("index.php");
            request
                .AddParameter("page", "dapi")
                .AddParameter("s", "post")
                .AddParameter("q", "index")
                .AddParameter("limit", limit)
                .AddParameter("tags", TagsQueryBuilder.BuildTagsQuery(rating, tags))
                .AddParameter("pid", page)
                .AddParameter("json", "1");
```